src/transformers/models/cohere/modeling_cohere.py

```diff
@@ -992,9 +973,12 @@ def _update_causal_mask(
                 return attention_mask
             return None
 
-        if self.config._attn_implementation == "sdpa":
-            # For SDPA, when possible, we will rely on its `is_causal` argument instead of its `attn_mask` argument,
-            # in order to dispatch on Flash Attention 2.
+        # For SDPA, when possible, we will rely on its `is_causal` argument instead of its `attn_mask` argument, in
+        # order to dispatch on Flash Attention 2. This feature is not compatible with static cache, as SDPA will fail
+        # to infer the attention mask.
+        past_seen_tokens = past_key_values.get_seq_length() if past_key_values is not None else 0
+        using_static_cache = isinstance(past_key_values, StaticCache)
+        if self.config._attn_implementation == "sdpa" and not using_static_cache:
```

Comment:
this is new, and since we pass cahce position, let's use cache_position[0]

Reply:
Agreed in theory, can't do in practice: breaks torch.fx tests 💔

Reply:
yeah thought so